### PR TITLE
fix: shutdown app on nats connection closed

### PR DESCRIPTION
**Pull request by Dav-14**
Fixes ENG-327

- gateway to caddy on nats closed connection; 
  - Add configurable graceperiod throught Configuration CRD
    - DEV ENV: 5Second
    - TODO: Staging: 1m
    - 
- fx.App shutdown on nats.ClosedHandler;

### CODE REVIEW

**Review by gfyrag — APPROVED** [review]

**Review by flemzord — APPROVED** [review]

**Review by paul-nicolas — APPROVED** [review]

File: components/operator/apis/stack/v1beta3/broker.go [hunk 1]
```diff
@@ -16,8 +16,17 @@ type KafkaConfig struct {
 }
 
 type NatsConfig struct {
-	URL string `json:"url"`
+	Hostname string `json:"hostname"`
```
Comment (gfyrag) [inline components/operator/apis/stack/v1beta3/broker.go:19]: You can't change already used CRDs.

File: components/operator/apis/stack/v1beta3/gateway_types.go [hunk 2]
```diff
@@ -11,4 +11,7 @@ type GatewaySpec struct {
 
 	// +optional
 	EnableAuditPlugin *bool `json:"enableAuditPlugin,omitempty"`
+
+	// +optional
```
Comment (gfyrag) [inline components/operator/apis/stack/v1beta3/gateway_types.go:15]: Seems to be useless, you already know the endpoint no?

File: components/operator/internal/modules/service.go [hunk 3]
```diff
@@ -326,6 +326,7 @@ type Service struct {
 	InjectPostgresVariables bool
 	HasVersionEndpoint      bool
 	Liveness                Liveness
+	LivenessEndpoint        *string
```
Comment (gfyrag) [inline components/operator/internal/modules/service.go:329]: Pointer is useless there, an empty string is enough to define than the endpoint is not overriden

**Review by gfyrag — APPROVED** [review]

File: components/operator/apis/stack/v1beta3/gateway_types.go (hunk 2 above)
Comment (Dav-14) [reply components/operator/apis/stack/v1beta3/gateway_types.go:15]: No i don't know the endpoint. 

In Nats case
I can guess the host name. 
But I need to hardcode monitoring port somewhere.

**Pull request merged**